ftva_lab_data/views.py

```diff
@@ -57,3 +62,54 @@ def view_item(request, item_id):
     item = SheetImport.objects.get(id=item_id)
 
     return render(request, "view_item.html", {"item": item})
+
+
+def view_items(request: HttpRequest) -> HttpResponse:
+    return render(request, "view_items.html", context={"columns": COLUMNS})
+
+
+def render_table(request: HttpRequest) -> HttpResponse:
+    """Handles search and pagination of table
+
+    Search can either be column-specific, determined by dropdown,
+    or broad, CTRL-F-style across all fields
+    """
+    display_fields = [field for field, _ in COLUMNS]
+
+    search = request.GET.get("search", "")
+    column = request.GET.get("column", "")
+    page = request.GET.get("page", 1)
+
+    # Only need display fields, plus ID for creating links
+    items = SheetImport.objects.only(*display_fields, "id").order_by("id")
+    if search:
+        if column and column in display_fields:
+            # Scoped search to selected column
+            items = items.filter(**{f"{column}__icontains": search})
+        else:
+            # General CTRL-F-style search across all configured fields
+            query = Q()  # start with empty Q() object, always True
+            for field in display_fields:  # then add queries for all valid fields
+                query |= Q(**{f"{field}__icontains": search})
+            items = items.filter(query)
+
+    paginator = Paginator(items, 10)  # TODO: make configurable
```

Comment:
The default of 10 seems reasonable.  If it's easy to add a UI element to let the user change this, go ahead... though if they'd have to change it again with every session, I'd rather not get into providing personal settings persistence at this point.
Otherwise, this `TODO` can be removed :)